### ISSUE TITLE
[version 1.44] No valid credential sources found for AWS Provider

### ISSUE THREAD

**Issue by cullenmcdermott** [terraform-providers/terraform-provider-aws#6462]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.10.8
```

### Affected Resource(s)

* provider.aws

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "${var.region}"
}
```

### Output
```
       Error: Error refreshing state: 1 error(s) occurred:

       * provider.aws: No valid credential sources found for AWS Provider.
       	Please see https://terraform.io/docs/providers/aws/index.html for more information on
       	providing credentials for the AWS Provider
```

### Expected Behavior

I expected terraform to respect my `AWS_PROFILE=saml` environment variable. 1.43.2 works but when I change my provider to 1.44.0 it no longer respects the variable. aws CLI and Packer both work fine so I know my credentials are good.

### Actual Behavior

The error above is produced

### Steps to Reproduce

1. `terraform apply`

### References

* #6320 seems related

**Comment by makennedy-clgx** [comment]
Reverting to release -1 works as expected.

**Comment by bflad** [comment]
Hi folks 👋 Similar to #6320 this may be related to upstream AWS Go SDK changes as I'm not aware to any changes in the provider authentication. My recommendation is to turn on [Terraform debug logging](https://www.terraform.io/docs/internals/debugging.html) in your environment (e.g. `TF_LOG=debug terraform plan`) to see if the underlying error(s) cannot be found instead of our less helpful error here.

The only change upstream in this regard I'm aware of is https://github.com/aws/aws-sdk-go/pull/2265

**Comment by cullenmcdermott** [comment]
Here's a gist with debug error output: https://gist.github.com/cullenmcdermott/9e2e7641fc3cc88117645e4e848901ff

**Comment by geekbass** [comment]
Same issue. Its ignoring the credentials in this release.
`2018-11-14T16:16:13.208-0500 [DEBUG] plugin.terraform-provider-aws_v1.44.0_x4: 2018/11/14 16:16:13 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2018/11/14 16:16:18 [ERROR] root.dcos: eval: *terraform.EvalConfigProvider, err: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
2018/11/14 16:16:18 [ERROR] root.dcos: eval: *terraform.EvalSequence, err: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
2018/11/14 16:16:18 [ERROR] root.dcos: eval: *terraform.EvalOpFilter, err: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
2018/11/14 16:16:18 [ERROR] root.dcos: eval: *terraform.EvalSequence, err: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
2018/11/14 16:16:18 [DEBUG] plugin: waiting for all plugin processes to complete...

Error: Error refreshing state: 1 error(s) occurred:

* module.dcos.provider.aws: No valid credential sources found for AWS Provider.
	Please see https://terraform.io/docs/providers/aws/index.html for more information on
	providing credentials for the AWS Provider
`

Reverting to previous version is fine.

**Comment by cullenmcdermott** [comment]
@bflad Not sure if this matters but in your output it looks like you are using v1.45.0

**Comment by atkinsj** [comment]
I'm experiencing this on `* provider.aws: version = "~> 1.45"` as well.

**Comment by bshelton229** [comment]
In our case we narrowed this down to the fact we were using "pretty" formatting in our ini file. We generate a credentials file using short-lived tokens via tooling. Updating the tooling to not us pretty spacing fixed the issue with the newer aws sdk.

Example ini format that stopped working with the upgrade (spacing not may remain pretty formatted in github)

```
[federated]
aws_access_key_id     = 123
aws_secret_access_key = 456
aws_session_token     = 789

[staging]
region         = us-east=1
role_arn       = arn:aws:iam::123456789:role/MyRole
source_profile = federated
```

Having the tooling stop doing pretty alignment allowed the new provider to work

```
[federated]
aws_access_key_id = 123
aws_secret_access_key = 456
aws_session_token = 789

[myprofile]
region = us-east-1
role_arn = arn:aws:iam::123456789:role/MyRole
source_profile = federated
```

Our tool was basically putting whitespace around the equals for alignment, which wasn't working with the new SDK.

**Comment by atkinsj** [comment]
Thanks @bshelton229, confirmed that resolves it for me as well.

**Comment by livingstaccato** [comment]
My team is using saml2aws and has run into this problem as well.

Can confirm that, on MacOS, `sed -i '' -E 's/ +/ /g' ~/.aws/credentials` fixed the issue.

**Comment by sstarcher** [comment]
Same as @livingstaccato we use saml2aws and this causes terraform to be non-functional.

**Comment by pradeepbhadani** [comment]
faced similar issue and @livingstaccato  solution fixed issue for me on Mac.

**Comment by makennedy-clgx** [comment]
Looks like their release is done. https://github.com/aws/aws-sdk-go/releases/tag/v1.15.79

**Comment by bflad** [comment]
The fix for this (AWS Go SDK update) is merged and will release with version 1.46.0 of the AWS provider, in a few minutes. 🚀

**Comment by bflad** [comment]
This has been released in [version 1.46.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1460-november-20-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
